qiskit/providers/ibmq/credentials/configrc.py

```diff
@@ -89,11 +98,30 @@ def write_qiskit_rc(
         filename: Full path to the configuration file. If ``None``, the default
             location is used (``$HOME/.qiskit/qiskitrc``).
     """
-    def _credentials_object_to_dict(obj: Credentials) -> Dict[str, Any]:
+    def _credentials_object_to_dict(credentials_obj: Credentials) -> Dict[str, Any]:
         """Convert a ``Credential`` object to a dictionary."""
-        return {key: getattr(obj, key) for key in
-                ['token', 'url', 'proxies', 'verify']
-                if getattr(obj, key)}
+        # Handle the keys/values that do not need special handling or conversion.
+        credentials_dict = {key: getattr(credentials_obj, key)
+                            for key in ['token', 'proxies', 'verify']
+                            if getattr(credentials_obj, key)}
+
+        # If a hgp is specified in `credentials_obj`, the `url` was modified to include hgp.
+        # As a result, save `base_url` to the account, rather than `url`.
+        credentials_dict['url'] = getattr(credentials_obj, 'base_url')
+
+        # Convert the hub/group/project to a dict, for easier handling.
+        hgp_dict = {
+            'hub': credentials_obj.hub,
+            'group': credentials_obj.group,
+            'project': credentials_obj.project
+        }
+
+        if all(hgp_dict.values()):
+            provider_to_save_as_str = get_provider_as_str(hgp_dict)
+            # Add the provider that will be saved to disk.
+            credentials_dict['default_provider'] = provider_to_save_as_str
+
+        return credentials_dict
 
     def _section_name(credentials_: Credentials) -> str:
```

Comment:
When specifying a default provider to save to disk, the ``base_name`` is given by the `{}_{}_{}_{}` formatter. 

It seems like this is set, in the case that multiple ``[ibmq*]`` sections are present. If multiple sections are allowed, it makes sense to keep this. Otherwise, we might be able to change it so that ``[ibmq]`` is written as the section name, for the account saved in disk?